airflow/assets/__init__.py

```diff
@@ -372,6 +372,26 @@ def iter_dag_dependencies(self, *, source: str, target: str) -> Iterator[DagDepe
         )
 
 
+class Dataset(Asset):
+    """Subclass of asset."""
+
+    group: str = "dataset"
+
+    def __init__(self, *args, **kwargs) -> None:
+        kwargs["group"] = Dataset.group
+        super().__init__(*args, **kwargs)
```

Comment:
This silently overrides `group`; not sure if that’s a good thing.

Maybe we should still allow the user to override group, with `dataset` only being a default? And maybe we should add a class-level variable on Asset and subclasses that’s not expected to be overridden (`asset_type` or something? similar to how each operator class has an `task_type`)

Reply:
I'm ok with it being a default. both sounds a big weird to me 🤔 

yep, asset_type would work. I guess the asset_type of the base asset could just be "asset"?

Reply:
just updated